### PR TITLE
Use meters as size scale in IconLayer and TextLayer

### PR DESCRIPTION
**Pull request by xintongxia**
For deck.gl 7.0 API audit

- `getSize` return size in meters (replace size in pixels)
- Add `sizeMinPixels` and `sizeMaxPixels`
- Fix `IconLayer` and `TextLayer` examples

Problems:
Change `size` to `meters` from `pixels` is breaking change, the old examples that use `IconLayer` are not working. I manually get the initial viewport parameter `metersPerPixel` and calculate the size in meters to match the original example. There are more examples I haven't changed yet. But is there other better way to make old examples work? Also other people whoever use `IconLayer` will be impacted by this change.

### CODE REVIEW

File: examples/website/icon/icon-cluster-layer.js [hunk 1]
```diff
@@ -64,9 +74,14 @@ export default class IconClusterLayer extends CompositeLayer {
...
         iconMapping,
         sizeScale,
+        sizeMinPixels,
+        sizeMaxPixels,
         getPosition,
         getIcon: d => d.zoomLevels[z] && d.zoomLevels[z].icon,
-        getSize: d => d.zoomLevels[z] && d.zoomLevels[z].size,
+        getSize: d => {
+          const size = d.zoomLevels[z] && d.zoomLevels[z].size;
+          return Number(size) * 18040;
```
Comment (Pessimistress) [inline examples/website/icon/icon-cluster-layer.js]: Add comment: what is this number? Use `sizeScale`?

File: modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer-vertex.glsl.js [hunk 2]
```diff
@@ -54,15 +56,20 @@ vec2 rotate_by_angle(vec2 vertex, float angle) {
   return rotationMatrix * vertex;
 }
 
-void main(void) {
+void main(void) {
```
Comment (Pessimistress) [inline modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer-vertex.glsl.js]: Extra space

File: modules/layers/src/text-layer/text-layer.js [hunk 3]
```diff
@@ -267,6 +271,8 @@ export default class TextLayer extends CompositeLayer {
         getPixelOffset: this._getAccessor(getPixelOffset),
         fp64,
         sizeScale: sizeScale * scale,
+        sizeMinPixels,
+        sizeMaxPixels,
```
Comment (Pessimistress) [inline modules/layers/src/text-layer/text-layer.js]: Should these two be adjusted with `scale` as well?

File: test/render/test-cases.js [hunk 4]
```diff
@@ -487,11 +487,13 @@ export const TEST_CASES = [
         data: dataSamples.points,
         iconAtlas: ICON_ATLAS,
         iconMapping: dataSamples.iconAtlas,
-        sizeScale: 12,
+        sizeScale: 12 * 21.34321226467833,
```
Comment (Pessimistress) [inline test/render/test-cases.js]: Just use a reasonable integer and then update the golden images.

File: modules/layers/src/icon-layer/icon-layer.js [hunk 5]
```diff
@@ -180,7 +183,10 @@ export default class IconLayer extends Layer {
         Object.assign({}, uniforms, {
           iconsTexture,
           iconsTextureDim: [iconsTexture.width, iconsTexture.height],
-          sizeScale
+          sizeScale,
+          sizeMinPixels,
+          sizeMaxPixels,
+          sizeInMeters: sizeUnits === 'meters'
```
Comment (Pessimistress) [inline modules/layers/src/icon-layer/icon-layer.js]: Just amend `sizeScale` here and avoid doing this in the shader for every vertex:

```
sizeScale: sizeScale * (sizeUnits === 'pixels' ? this.context.viewport.distanceScales.metersPerPixel[2] : 1)
```

File: test/render/test-cases.js [hunk 6]
```diff
@@ -572,6 +572,42 @@ export const TEST_CASES = [
...
+    layers: [
+      new IconLayer({
+        id: 'icon-lnglat',
+        data: dataSamples.points,
+        iconAtlas: ICON_ATLAS,
+        iconMapping: dataSamples.iconAtlas,
+        sizeScale: 256,
+        sizeUnits: 'meters',
+        sizeMinPixels: 1,
+        sizeMaxPixels: 2048,
```
Comment (xintongxia) [inline test/render/test-cases.js]: delete `sizeMinPixels` and `sizeMaxPixels`

File: examples/layer-browser/src/examples/core-layers.js [hunk 7]
```diff
@@ -63,11 +63,14 @@ const IconLayerExample = {
   props: {
     iconAtlas: 'data/icon-atlas.png',
     iconMapping: dataSamples.iconAtlas,
-    sizeScale: 24,
     getPosition: d => d.COORDINATES,
     getColor: d => [64, 64, 72],
     getIcon: d => (d.PLACEMENT === 'SW' ? 'marker' : 'marker-warning'),
     getSize: d => (d.RACKS > 2 ? 2 : 1),
+    sizeMinPixels: 1,
```
Comment (xintongxia) [inline examples/layer-browser/src/examples/core-layers.js]: revert changes for layer-browser

File: modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer.js [hunk 8]
```diff
@@ -28,6 +28,10 @@ const DEFAULT_GAMMA = 0.2;
 const DEFAULT_BUFFER = 192.0 / 256;
 
 const defaultProps = {
+  sizeUnits: 'pixels',
+  sizeMinPixels: 0,
+  sizeMaxPixels: Number.MAX_SAFE_INTEGER,
```
Comment (Pessimistress) [inline modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer.js]: These are already present in the base class (IconLayer)'s defaultProps

File: modules/layers/src/icon-layer/icon-layer-vertex.glsl.js [hunk 9]
```diff
@@ -50,12 +52,18 @@ vec2 rotate_by_angle(vec2 vertex, float angle) {
 
 void main(void) {
   vec2 iconSize = instanceIconFrames.zw;
+  // convert size in meters to pixels, then scaled and clamp
+  
+  float sizeScalePixels = instanceSizes * sizeScale;
+  
+  sizeScalePixels = clamp(sizeScalePixels, sizeMinPixels, sizeMaxPixels);
```
Comment (Pessimistress) [inline modules/layers/src/icon-layer/icon-layer-vertex.glsl.js]: You still need to call `project_scale` here

File: modules/layers/src/icon-layer/icon-layer-vertex.glsl.js [hunk 10]
```diff
@@ -50,12 +52,19 @@ vec2 rotate_by_angle(vec2 vertex, float angle) {
 
 void main(void) {
   vec2 iconSize = instanceIconFrames.zw;
+  // convert size in meters to pixels, then scaled and clamp
+  
+  float sizeScalePixels = clamp(
```
Comment (Pessimistress) [inline modules/layers/src/icon-layer/icon-layer-vertex.glsl.js]: Nit: call it `sizePixels` as it's not a scale

File: modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer-vertex.glsl.js [hunk 11]
```diff
@@ -56,13 +58,20 @@ vec2 rotate_by_angle(vec2 vertex, float angle) {
 
 void main(void) {
   vec2 iconSize = instanceIconFrames.zw;
+  // convert size in meters to pixels, then scaled and clamp
+  
+  float sizeScalePixels = clamp(
```
Comment (Pessimistress) [inline modules/layers/src/text-layer/multi-icon-layer/multi-icon-layer-vertex.glsl.js]: Ditto

**Pull request merged**